functions/Find-DbaCommand.ps1

```diff
@@ -0,0 +1,146 @@
+Function Find-DbaCommand
+{
+<#
+.SYNOPSIS
+Finds dbatools commands searching through the inline help text
+
+.DESCRIPTION
+Finds dbatools commands searching through the inline help text, building a consolidated json index and querying it because Get-Help is too slow
+
+.PARAMETER Tag
+Finds all commands tagged with this tag
+
+.NOTES
+Tags: find
+Author: niphlod
+
+
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Find-DbaCommand
+
+.EXAMPLE
+Find-DbaCommand "snapshot"
+
+For lazy typers: finds all commands searching the entire help for "snapshot"
+
+.EXAMPLE
+Find-DbaCommand -Pattern "snapshot"
+
+For rigorous typers: finds all commands searching the entire help for "snapshot"
+
+.EXAMPLE
+Find-DbaCommand -Tag copy
+
+Finds all commands tagged with "copy"
+
+.EXAMPLE
+Find-DbaCommand -Author "chrissy"
+
+Finds every command whose author contains our beloved "chrissy"
+
+.EXAMPLE
+Find-DbaCommand -Pattern "snapshot" -Rebuild
+
+Finds all commands searching the entire help for "snapshot", rebuilding the index (good for developers)
+#>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    Param (
+        [String]$Pattern,
+        [String]$Tag,
+        [String]$Author,
+        [switch]$Rebuild
+    )
+    BEGIN
+    {
+        $tagsRex = ([regex]'(?m)^[\s]{0,15}Tags:(.*)$')
+        $authorRex = ([regex]'(?m)^[\s]{0,15}Author:(.*)$')
+
+        function Get-DbaHelp([String]$commandname) {
+            $thishelp = Get-Help $commandname -Full
+            $thebase = @{}
+            $thebase.CommandName = $commandname
+            $thebase.Name = $thishelp.name
+
+            ## fetch the description
+            $thebase.Description = $thishelp.Description.text
+
+            ## fetch examples
+            $thebase.Examples = $thishelp.Examples | Out-String -Width 120
+
+            ## fetch help link
+            $thebase.Links = ($thishelp.relatedLinks).navigationLink.uri
+
+            ## fetch the synopsis
+            $thebase.Synopsis = $thishelp.Synopsis
+
+            ## store notes
+            $as = $thishelp.alertSet | Out-String -Width 120
+
+            ## fetch the tags
+            $tags = $tagsrex.Match($as).Groups[1].Value
+            if($tags) {
+                $thebase.Tags = $tags.Trim().Split(',')
```

Comment:
I'd split first and trim then.
That way, tags can safely have a whitespace behind the separating comma.